src/components/standalone/hotspot/SettingsContent.vue

```diff
@@ -0,0 +1,695 @@
+
+interface Login {
+  hostname: string
+  username: string
+  password: string
+}
+
+const loginForm = ref<Login>({
+  hostname: 'my.nethspot.com',
+  username: '',
+  password: ''
+})
+
+interface Configuration {
+  parentHotspot: string
+  unitName: string
+  unitDescription: string
+  networkDevice: string
+  networkAddress: string
+  dhcpRangeStart: string
+  dhcpRangeEnd: string
+  maxClientsAllowed: string
+}
+
+interface ParentHotspot {
+  id: string
+  label: string
+  name: string
+}
+
+interface NetworkDevice {
+  id: string
+  name: string
+  label: string
+}
+
+const configurationForm = ref<Configuration>({
+  parentHotspot: '',
+  unitName: '',
+  unitDescription: '',
+  networkDevice: '',
+  networkAddress: '',
+  dhcpRangeStart: '',
+  dhcpRangeEnd: '',
+  maxClientsAllowed: ''
+})
+
+let isLoggedIn = ref(false)
+let activeConfiguration = ref(false)
+let loading = ref(false)
+let logging = ref(false)
+let loadingUnregister = ref(false)
+let loadingDhcpRange = ref(false)
+let showUnregisterModal = ref(false)
+let emptyDevices = ref(false)
+let saving = ref(false)
+let successSaving = ref(false)
+let hostnameRef = ref()
+let usernameRef = ref()
+let passwordRef = ref()
+let unitNameRef = ref()
+let unitDescriptionRef = ref()
+let networkAddressRef = ref()
+let dhcpRangeStartRef = ref()
+let dhcpRangeEndRef = ref()
+let parentHotspotList = ref<Array<ParentHotspot>>()
+let networkDeviceList = ref<Array<NetworkDevice>>()
+
+let objError = {
+  notificationTitle: '',
+  notificationDescription: '',
+  hostname: '',
+  username: '',
+  password: '',
+  unitName: '',
+  unitDescription: '',
+  networkAddress: '',
+  dhcpRangeStart: '',
+  dhcpRangeEnd: '',
+  maxClientsAllowed: ''
+}
+let error = ref({ ...objError })
+let errorLoadingData = ref({ ...objError })
+let errorSave = ref({ ...objError })
+let errorUnregister = ref({ ...objError })
+let errorDhcpRange = ref({ ...objError })
+
+onMounted(() => {
+  getFirewallData()
+  getConfiguration()
+})
+
+async function getFirewallData() {
+  loading.value = true
+  errorLoadingData.value = { ...objError }
+
+  // Retrieve parent hotspot
+  try {
+    let getParentHotspot = await ubusCall('ns.dedalo', 'list-parents', {})
+    if (
+      getParentHotspot &&
+      getParentHotspot.data &&
+      getParentHotspot.data.parents &&
+      getParentHotspot.data.parents.length
+    ) {
+      isLoggedIn.value = true
+      parentHotspotList.value = getParentHotspot.data.parents.map((item: ParentHotspot) => ({
+        id: item.id,
+        label: item.name
+      }))
+      configurationForm.value.parentHotspot = getParentHotspot.data.parents[0].id
+    }
+  } catch (exception: any) {
+    errorLoadingData.value.notificationTitle = t('error.cannot_retrieve_parent_hotspot')
+    errorLoadingData.value.notificationDescription = t(getAxiosErrorMessage(exception))
+  } finally {
+    try {
+      let getNetworkDevice = await ubusCall('ns.dedalo', 'list-devices', {})
```

Comment:
If there are no dependencies between `list-parents` and `list-devices` we can invoke them at the same time. We can split `getFirewallData` function and invoke the two functions inside `onMouted`